### PR TITLE
Plugins: Disable uninstall while cloud uninstall is not completed

### PR DESCRIPTION
**Pull request by oshirohugo**
**What is this feature?**

Disable the uninstall button while the cloud uninstall is not completed.

**Why do we need this feature?**

This is necessary so the user is not able to uninstall a plugin that is being removed.

**Who is this feature for?**

Plugin catalog users

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.tsx [hunk 1]
```diff
@@ -126,7 +131,7 @@ export function InstallControlsButton({
           onDismiss={hideConfirmModal}
         />
         <HorizontalGroup align="flex-start" width="auto" height="auto">
-          <Button variant="destructive" disabled={isUninstalling} onClick={showConfirmModal}>
+          <Button variant="destructive" disabled={isUninstalling || isUninstallingInCloud} onClick={showConfirmModal}>
```
Comment (academo) [inline public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.tsx:134]: This expression along the previous one (line 117) makes the logic to show `isUninstalling` hard to follow because the logic is now spread in 3 different places.

Could you make a single boolean variable that holds the final calculated value and it is used here?

File: public/app/features/plugins/admin/helpers.ts [hunk 2]
```diff
@@ -51,6 +51,8 @@ export function mergeLocalsAndRemotes({
           ? true
           : instancesSet.has(remotePlugin.slug) && catalogPlugin.isInstalled;
         catalogPlugin.isInstalled = instancesSet.has(remotePlugin.slug) || catalogPlugin.isInstalled;
+
+        catalogPlugin.isUninstalling = Boolean(localCounterpart) && !instancesSet.has(remotePlugin.slug);
```
Comment (leventebalogh) [inline public/app/features/plugins/admin/helpers.ts:55]: I think this name could be a bit misleading in OSS / On-prem (if the plugin is simply installed on-prem, then this should evaluate to `true`, as `!instancesSet.has(remotePlugin.slug)` will be `true`, right?)

Naming it something like `isUninstallingFromRemote()` or `isUninstallingFromInstance()` could be more clear I think. What do you think?

File: public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.tsx [hunk 3]
```diff
@@ -114,6 +114,11 @@ export function InstallControlsButton({
   };
 
   if (pluginStatus === PluginStatus.UNINSTALL) {
+    const disableUninstall =
+      config.pluginAdminExternalManageEnabled && configCore.featureToggles.managedPluginsInstall
+        ? plugin.isUninstalling
+        : isUninstalling;
```
Comment (leventebalogh) [inline public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.tsx:120]: I think it would worth updating the tests as well to cover this scenario in `InstallControlsButton.test.tsx`.

**Review by leventebalogh — CHANGES_REQUESTED** [review]

**Comment by oshirohugo** [comment]
@leventebalogh could you please take another look?

**Review by leventebalogh — APPROVED** [review]
LGTM 🚀
